src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleConnectionParameters.java

```diff
@@ -114,6 +135,90 @@ private String generateUrl() {
       buf.append("/");
       buf.append(serviceName);
     }
+
     return buf.toString();
   }
+
+  private String generateSecureUrl(StringBuilder buf) {
+    return new OracleTNSEntryBuilder()
+        .withProtocol("TCPS")
+        .withHost(host)
+        .withPort(port)
+        .withInstanceName(instance)
+        .withServiceName(serviceName)
+        .build(buf);
+  }
+
+  private void checkInstanceAndServiceName() {
+    if (instance != null && serviceName != null) {
+      String errorMessage = "Instance (SID) : [" + instance + "] and Service Name : [" + serviceName
+          + "] were provided at the same time, please configure only one";
+
+      throw new ModuleException(errorMessage, CONNECTIVITY, new ConnectionException(errorMessage));
```

Comment:
why not just throwing the ConnectionException directly?

Reply:
The signature does not have an Exception **_throw_** clause, so the method relied on ModuleException (a RuntimeException, an unchecked exception).
Is there any way to get around this?

Reply:
I think the actual problem is that you're putting connectivity logic in a class which is only meant to hold parameters. I think the problem is that this logic actually belongs in the connection provider.